upgrade/v2.5/script/upgrade.sh

```diff
@@ -0,0 +1,119 @@
+    echo "No IPPools found to migrate."
+    echo "Moving on..."
+else
+    echo "Migrating the following IPPools..."
+    cat ippool.yaml
+    echo
+    /sbin/calicoctl-v1.5 apply -f ippool.yaml
+    if [ $? != 0 ]; then
+        echo "Failed to apply the IPPools"
+        exit 1
+    else
+        echo "Successfully migrated the IPPools!"
+    fi
+  fi
+fi
+
+# Get BGPPeers list with old calicoctl (from TPR) and save it as a yaml file.
+# Apply the yaml file with the new calicoctl (to CRD)
+echo; echo "Migrating BGPPeers..."
+/sbin/calicoctl-v1.4 get bgppeers -o yaml > bgppeers.yaml
+if [ $? != 0 ]; then
+    echo "Failed to get Global BGP Peers through calicoctl"
+    exit 1
+else if [ `cat ippool.yaml | wc -l` == 1 ]; then
+    echo "No BGPPeers found to migrate."
+    echo "Moving on..."
+else
+    echo "Migrating the following BGPPeers..."
+    cat bgppeers.yaml
+    echo
+    /sbin/calicoctl-v1.5 apply -f bgppeers.yaml
+    if [ $? != 0 ]; then
+        echo "Failed to apply the BGPPeers"
+        exit 1
+    else
+        echo "Successfully migrated the BGPPeers!"
+    fi
+  fi
+fi
+
+# List all the Felix configs using kubectl and save it in a yaml file.
+# Change the apiVersion from 'projectcalico.org/v1' to 'crd.projectcalico.org/v1',
+# rename resource kind from 'GlobalConfig' to 'GlobalFelixConfig' and save it in a new yaml file.
+# Apply the modified yaml file.
+echo; echo "Migrating GlobalFelixConfig..."
+/sbin/kubectl get globalconfig --all-namespaces -o yaml > tpr-felixconfig.yaml
+if [ $? != 0 ]; then
+    echo "Failed to get Global Felix config through Kubectl"
+    exit 1
+else
+    echo "Successfully got the Felix Config:"
+    cat tpr-felixconfig.yaml
+    cat tpr-felixconfig.yaml | grep 'items\: \[\]'
+    if [ $? == 0 ]; then
+        echo "No GlobalFelixConfig found to migrate."
+        echo "Moving on..."
+    else
+        cat tpr-felixconfig.yaml | sed '/apiVersion/s/projectcalico\.org\/v1/crd\.projectcalico\.org\/v1/g' | sed '/kind/s/GlobalConfig/GlobalFelixConfig/g' > crd-felixconfig.yaml
+        echo "Successfully renamed the resource kind"
+        cat crd-felixconfig.yaml
+        /sbin/kubectl apply -f crd-felixconfig.yaml
+        if [ $? != 0 ]; then
+            echo "Failed to apply the GlobalFelixConfig"
+            exit 1
+        else
+            echo "Successfully migrated the GlobalFelixConfig!"
+        fi
+    fi
+fi
+
+
+# List all the BGP configs using kubectl and save it in a yaml file.
+# Change the apiVersion from 'projectcalico.org/v1' to 'crd.projectcalico.org/v1' and save it in a new yaml file.
+# Apply the modified yaml file.
+echo; echo "Migrating GlobalBGPConfig..."
+/sbin/kubectl get globalbgpconfig --all-namespaces -o yaml > tpr-bgpconfig.yaml
+if [ $? != 0 ]; then
+    echo "Failed to get the Global BGP config through Kubectl"
+    exit 1
+else
+    echo "Successfully got the Global BGP Config:"
+    cat tpr-bgpconfig.yaml
+    cat tpr-bgpconfig.yaml | grep 'items\: \[\]'
```

Comment:
does this output to the logs?  

you could assign to a var to stifle that, e.g. `d=grep 'blah'`

Also, you don't need cat here either since grep takes an optional file parameter.